modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -312,34 +313,35 @@ object EnrichmentManager {
   }
 
   /** Initialize the mutable [[EnrichedEvent]]. */
-  private def setupEnrichedEvent(
+  private def setupEnrichedEvent[F[_]: Monad](
     raw: RawEvent,
     e: EnrichedEvent,
     etlTstamp: DateTime,
     processor: Processor
-  ): Either[FailureDetails.SchemaViolation, Unit] = {
-    e.event_id = EE.generateEventId() // May be updated later if we have an `eid` parameter
-    e.v_collector = raw.source.name // May be updated later if we have a `cv` parameter
-    e.v_etl = ME.etlVersion(processor)
-    e.etl_tstamp = EE.toTimestamp(etlTstamp)
-    e.network_userid = raw.context.userId.map(_.toString).orNull // May be updated later by 'nuid'
-    e.user_ipaddress = ME
-      .extractIp("user_ipaddress", raw.context.ipAddress.orNull)
-      .toOption
-      .orNull // May be updated later by 'ip'
-    // May be updated later if we have a `ua` parameter
-    setUseragent(e, raw.context.useragent)
-    // Validate that the collectorTstamp exists and is Redshift-compatible
-    val collectorTstamp = setCollectorTstamp(e, raw.context.timestamp).toValidatedNel
-    // Map/validate/transform input fields to enriched event fields
-    val transformed = Transform.transform(raw, e)
-
-    (collectorTstamp |+| transformed)
-      .leftMap(AtomicFields.atomicErrorsToSchemaViolation)
-      .toEither
-  }
+  ): F[Either[FailureDetails.SchemaViolation, Unit]] =
+    Monad[F].pure {
```

Comment:
Is there a reason you went for `Monad[F].pure` instead of `Sync[F].delay`?  It's fine if the answer is because that would be too much of a change, to introduce a `Sync` into this part of the code.

Reply:
Indeed it's because only `Monad` was available here and I didn't see a point in introducing `Sync`. What would the advantage be? It's not a big change to introduce it (we'll very likely end up needing `Sync` at some point anyway)

Reply:
Someone who does all that would really try to break Enrich on purpose :joy: Tests would break anyway, but yeah let's add a bit of FP good practices, even though we have this disgusting mutable event, I'll introduce `Sync`

Reply:
Now I'm really tempted to wrap all the places of Enrich that mutate the event inside `Sync[F].delay()`, but making Enrich more functional is outside of this task